src/encode.cpp

```diff
@@ -974,6 +974,31 @@ void encodeOp(State &st, mlir::tosa::TransposeOp op, bool) {
 
 }
 
+template<>
+void encodeOp(State &st, mlir::tosa::GatherOp op, bool) {
+  // values, output - 3D dimension, indices - 2D dimension.
+  // These were checked by default MLIR verifier
+
+  // input's dim sizes = [N, K, C]
+  auto values = st.regs.get<Tensor>(op.values());
+  // indices's dim sizes = [N, W]
+  auto indices = st.regs.get<Tensor>(op.indices());
+  // output tensor dim = [N, W, C]
+  // output[n][w][c] = values[n][indices[n][w]][c]
+  vector<Expr> outputDims = {values.getDim(0), indices.getDim(1), values.getDim(2)};
+  vector<Expr> indVars = Index::boundIndexVars(outputDims.size());
+  auto [idx, idxInbounds] = indices.get({indVars[0], indVars[1]});
+  auto [outputValue, inputInbounds] = values.get({indVars[0], idx, indVars[2]});
+
+  st.wellDefined(op, move(idxInbounds));
+  st.wellDefined(op, move(inputInbounds));
```

Comment:
These two `st.wellDefined` statements are invalid because `idxInbounds` and `inputInbounds` are using bound variables.
`idxInbounds` and `inputInbounds` must have appeared inside a forall statement.